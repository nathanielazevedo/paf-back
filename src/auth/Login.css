/** @format */

.login-container{
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: #1c2227;
}

.login-form{
  display: flex;
  flex-direction: column;
  padding: 50px;
  background-color: #121212;
}

.login-title{
  color: white;
}

.login-input{
  background-color: transparent;
  border: none;
  border-bottom: solid white 1px;
  margin: 20px 0;
  color: white;
}