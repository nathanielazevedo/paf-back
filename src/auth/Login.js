/** @format */

import React from "react";
import "./Login.css";
import Form from "../form/Form";

//Loginfunc is passed down from App.js as a prop. This component passes the loginfunc down to the form.

function Login({ loginFunc }) {
  return (
    <div className="login-container">
      <Form title="Login" inputs={["username", "password"]} func={loginFunc} close={null}/>
    </div>
  );
}

export default Login;
