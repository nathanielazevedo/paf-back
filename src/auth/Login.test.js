/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Login from "./Login";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Login />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Login />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays expected text", function () {
  const { getAllByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Login />
      </UserProvider>
    </MemoryRouter>
  );

  expect(getAllByText("Login")[0]).toBeInTheDocument();
});
