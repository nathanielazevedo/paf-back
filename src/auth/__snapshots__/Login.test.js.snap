// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="login-container"
  >
    <div
      class="form-container"
    >
      <div
        class="form-inner"
      >
        <h3
          class="form-title"
        >
          Login
        </h3>
        <form
          class="form-input-container"
        >
          <div
            class="form-input-group"
          >
            <label
              for="username"
            >
              username
            </label>
            <input
              class="form-input"
              data-testid="username"
              name="username"
              placeholder="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-input-group"
          >
            <label
              for="password"
            >
              password
            </label>
            <input
              class="form-input"
              data-testid="password"
              name="password"
              placeholder="password"
              required=""
              type="text"
              value=""
            />
          </div>
          <button
            class="form-button"
            data-testid="button"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
